src/local_scheduler/local_scheduler_algorithm.cc

```diff
@@ -1281,9 +1357,15 @@ void handle_task_scheduled(LocalSchedulerState *state,
    * the database. */
   RAY_CHECK(state->db != NULL);
   RAY_CHECK(state->config.global_scheduler_exists);
-  /* Push the task to the appropriate queue. */
-  queue_task_locally(state, algorithm_state, execution_spec, true);
-  dispatch_tasks(state, algorithm_state);
+
+  if (state->static_resources["CPU"] == 0) {
+    // Give the task to the global scheduler to schedule.
+    give_task_to_global_scheduler(state, algorithm_state, execution_spec);
```

Comment:
Is this statement reachable if the global scheduler never assigns actor creation tasks to a node with no CPUs? If we can, let's try to avoid the `state->static_resources["CPU"] == 0` checks since they're a little cryptic if you don't know about the special case for actor resources.

Reply:
Yeah I think it's not possible. What if I change this to

```
// The global scheduler should never assign a task to a machine with 0 CPUs.
RAY_CHECK(state->static_resources["CPU"] != 0)
```

Reply:
Sure, that sounds good.